srtcore/queue.cpp

```diff
@@ -960,34 +960,24 @@ void CRendezvousQueue::updateConnStatus(EReadStatus rst, EConnectStatus cst, con
                 << "). removing from queue");
             // connection timer expired, acknowledge app via epoll
             i->m_pUDT->m_bConnecting = false;
-            if (!is_zero(i->m_tsTTL))
+            int ccerror = SRT_ECONNREJ;
+            if (i->m_pUDT->m_RejectReason == SRT_REJ_UNKNOWN)
             {
-                // Timer expired, set TIMEOUT forcefully
-                i->m_pUDT->m_RejectReason = SRT_REJ_TIMEOUT;
+                if (!is_zero(i->m_tsTTL))
+                {
+                    // Timer expired, set TIMEOUT forcefully
+                    i->m_pUDT->m_RejectReason = SRT_REJ_TIMEOUT;
+                    ccerror = SRT_ENOSERVER;
```

Comment:
This is a useful fix. Previously it was possible that a connection had already been rejected with some rejection code, but the rejection reason was overwritten with `SRT_REJ_TIMEOUT` if this timer hits in parallel.

Reply:
Yes, and that's exactly the fix. Jean reported that in several situations, tho it wasn't recorded as an issue.